### PR TITLE
Add Enum to new compiler rewrite.

### PR DESCRIPTION
**Pull request by djkevincr**

### CODE REVIEW

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java [hunk 1]
```diff
@@ -573,6 +578,23 @@ public void endStructDef(String identifier) {
...
+
+    public void endEnumDef(String identifier) {
+        EnumNode enumNode = this.enumStack.pop();
+        enumNode.setName(this.createIdentifier(identifier));
+        this.identifierNodeListStack.pop().forEach(identifierNode -> enumNode.addEnumField(identifierNode));
+        this.compUnit.addTopLevelNode(enumNode);
+    }
+
+    public void addEnumFieldList(List<String> enumFieldList) {
+        List<IdentifierNode> identifierList = new ArrayList<>();
```
Comment (sameera-jayasoma) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java]: Do we need a Stack of List of identifiers here?

Comment (djkevincr) [reply modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java]: +1

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java [hunk 2]
```diff
@@ -573,6 +578,23 @@ public void endStructDef(String identifier) {
         this.varListStack.pop().forEach(structNode::addField);
         this.compUnit.addTopLevelNode(structNode);
     }
+
+    public void startEnumDef() {
+        this.enumStack.add(TreeBuilder.createEnumNode());
+    }
+
+    public void endEnumDef(String identifier) {
```
Comment (sameera-jayasoma) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java]: Please add the diagnostic position to the enum node :)

Comment (djkevincr) [reply modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java]: +1

**Pull request merged**